scripts/cloud-check-control/cloudAtlasCheckControl.py

```diff
@@ -216,22 +225,21 @@ def atlasCheck(self):
 
             logger.info("Submitting spark job: {}".format(cmd))
             if self.ssh_cmd(cmd):
-                finish("Unable to execute spark job", -1)
+                self.finish("Unable to execute spark job", -1)
             # make sure spark job has started before checking for completion
             time.sleep(5)
-        else:
-            logger.info("Detected a running atlas check spark job.")
 
-        logger.info("About to wait for remote script to complete. If "
-                    "disconnected before the script completes then execute the "
-                    "following command to continue waiting:\n {} --id={}"
-                    .format(" ".join(sys.argv), self.instanceId))
-        # wait for script to complete
-        if self.wait_for_process_to_complete():
-            finish(
-                "Timeout waiting for script to complete. TODO - instructions to reconnect.",
-                -1,
-            )
+        if self.is_process_running("SparkSubmit"):
+            logger.info("About to wait for remote script to complete. If "
+                        "disconnected before the script completes then execute the "
+                        "following command to continue waiting:\n {} --id={}"
+                        .format(" ".join(sys.argv), self.instanceId))
+            # wait for script to complete
+            if self.wait_for_process_to_complete():
+                self.finish(
+                    "Timeout waiting for script to complete. TODO - instructions to reconnect.",
```

Comment:
Can --force be used if there was a timeout?

Reply:
yes. If there was a timeout and the internal script completed then the user would need to use force to rerun the script... or if force is not used then it would simply let the user know that the script completed and then attempt to sync the data produced to the s3 output.